### PR TITLE
feat(types): improve and test encoding of `KIND`

### PR DESCRIPTION
**Pull request by grtlr**
## Linked Issues

None

## Notes to Reviewer

As a reviewer, please pay particular attention to the following areas when reviewing this PR and tick the above boxes after you have completed the steps.

#### Test cases
* [ ] Review and run tests that were added/changed.
* [ ] Suggest places that may benefit from test cases.

### CODE REVIEW

File: src/db/collections/mod.rs [hunk 1]
```diff
@@ -32,56 +32,34 @@ pub use self::{
...
 };
 
 /// Helper to specify a kind for an output type.
-pub trait OutputKind {
+pub trait OutputKindQuery {
     /// Gets the output kind.
-    fn kind() -> Option<&'static str> {
-        None
-    }
+    fn kind() -> Option<&'static str>;
```
Comment (Alex6323) [inline src/db/collections/mod.rs:41]: why not simply make this a `const KIND: Option<&'static str>;` ?

Comment (Alex6323) [inline src/db/collections/mod.rs:41]: what about making this:
```rust
const KIND: &'static str;
```
and let `Output` KIND be `""`?

Comment (grtlr) [reply src/db/collections/mod.rs:41]: No that won't work with the query building.

**Review by Alexandcoats — APPROVED** [review]
much improvement, very wow

**Pull request merged**